packages/nimble-components/src/icon-base/tests/icons.stories.ts

```diff
@@ -1,22 +1,33 @@
 import type { Meta, StoryObj } from '@storybook/html';
-import { html, repeat } from '@microsoft/fast-element';
+import { html, ref, repeat } from '@microsoft/fast-element';
 import { DesignSystem } from '@microsoft/fast-foundation';
 import * as nimbleIconComponentsMap from '../../icons/all-icons';
 import { IconSeverity } from '../types';
 import {
     createUserSelectedThemeStory,
     overrideWarning
 } from '../../utilities/tests/storybook';
-import type { Icon } from '..';
 import {
     tokenNames,
     scssInternalPropertySetterMarkdown
 } from '../../theme-provider/design-token-names';
+import { Table, tableTag } from '../../table';
+import { tableColumnIconTag } from '../../table-column/icon';
+import { mappingIconTag } from '../../mapping/icon';
+import { tableColumnTextTag } from '../../table-column/text';
+import { iconMetadata } from './icon-metadata';
 
-const nimbleIconComponents = Object.values(nimbleIconComponentsMap);
+type IconName = keyof typeof nimbleIconComponentsMap;
+const data = Object.values(nimbleIconComponentsMap).map(klass => ({
+    tag: DesignSystem.tagFor(klass),
+    metaphor: iconMetadata[klass.name as IconName].tags.join(', ')
+}));
+
+type Data = (typeof data)[number];
 
 interface IconArgs {
     severity: keyof typeof IconSeverity;
+    tableRef: Table<Data>;
 }
 
 const metadata: Meta<IconArgs> = {
```

Comment:
I was finding it slightly jarring to see a huge table with no context. How about a bit of docs above it like this?
```suggestion
const metadata: Meta<IconArgs> = {
    description: 'Nimble icons can be slotted into other components or used independently. Each icon is available as a custom element. For example, `<nimble-icon-add></nimble-icon-add>`'.
```